homeassistant/components/sunweg/__init__.py

```diff
@@ -0,0 +1,190 @@
+"""The Sun WEG inverter sensor integration."""
+import datetime
+import json
+import logging
+
+from sunweg.api import APIHelper
+from sunweg.plant import Plant
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import StateType
+from homeassistant.util import Throttle
+
+from .const import CONF_PLANT_ID, DOMAIN, PLATFORMS
+from .sensor_types.sensor_entity_description import SunWEGSensorEntityDescription
+
+SCAN_INTERVAL = datetime.timedelta(minutes=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: config_entries.ConfigEntry
+) -> bool:
+    """Load the saved entities."""
+    api = APIHelper(entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
+    if not await hass.async_add_executor_job(api.authenticate):
+        _LOGGER.error("Username or Password may be incorrect!")
```

Comment:
you can trigger a reauth flow here so the user can update their username and/or password. I would do that in a follow up PR though

Reply:
I prefer to leave that for another PR